### ISSUE TITLE
False positives for google_checks when using switch expression in lambda

### ISSUE THREAD

**Issue by Edarke** [checkstyle/checkstyle#17120]
I have read check documentation: https://checkstyle.sourceforge.io/checks/misc/indentation.html#Indentation
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```bash
/var/tmp $ javac Example.java
/var/tmp $ java -jar ~/Downloads/google-java-format-1.26.0-all-deps.jar --replace Example.java; cat Example.java 
import java.util.Optional;

public class Example {

  /** Main method. */
  public static String main(Optional<Integer> opt) {
    return opt.map(
            n ->
                switch (n) {
                  case 1 -> "Test";
                  default -> "Default";
                })
        .orElse("");
  }
}


/var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"
/var/tmp $  java -jar ~/Downloads/google-java-format-1.27.0-all-deps.jar --replace Example.java; cat Example.java;  java $RUN_LOCALE -jar ~/Downloads/checkstyle-10.24.0-all.jar -c /google_checks.xml Example.java 
Starting audit...
[WARN] /Users/evan.darke/Downloads/Example.java:10:17: 'switch' has incorrect indentation level 16, expected level should be one of the following: 10, 12. [Indentation]
[WARN] /Users/evan.darke/Downloads/Example.java:11:19: 'case' child has incorrect indentation level 18, expected level should be one of the following: 12, 14. [Indentation]
[WARN] /Users/evan.darke/Downloads/Example.java:12:19: 'case' child has incorrect indentation level 18, expected level should be one of the following: 12, 14. [Indentation]
[WARN] /Users/evan.darke/Downloads/Example.java:13:17: 'switch rcurly' has incorrect indentation level 16, expected level should be one of the following: 10, 12. [Indentation]
Audit done.
```

**Comment by christophpre** [comment]
Having a similar issue when using a combined switch statement:

```java
switch (value) {
    case VALUE_1,
         VALUE_2, // CheckStyle ERROR
         VALUE_3 -> "Test"; //CheckStyle ERROR
    default -> "Default";
}
```

Since upgrading from 10.23.1 to 10.24.0, this was no problem before.

Could it be that this was introduced with the bugfix here: https://github.com/checkstyle/checkstyle/issues/15098 ?

**Comment by romani** [comment]
@Aziz-755 , do you see a way to do quick followup fix to ease problem on user side ?

**Comment by christophpre** [comment]
@romani I realized that I had different indentation settings than the current official google style. Everything else is the same. But imho, the issue is still there.

Created a Gist for you to test: https://gist.github.com/christophpre/9637c298e8516099c6010c03bc64c3f5

```bash
java -jar ./checkstyle-10.24.0-all.jar -c ./google_checks_adapted.xml Example.java
Starting audit...
[WARN] Example.java:17:18: 'case' child has incorrect indentation level 17, expected level should be 20. [Indentation]
[WARN] Example.java:18:18: 'case' child has incorrect indentation level 17, expected level should be 20. [Indentation]
Audit done.

java -jar ./checkstyle-10.23.0-all.jar -c ./google_checks.xml Example.java
Starting audit...
Audit done.
```

**Comment by romani** [comment]
@christophpre, please do us a favor and show by CLI
That if you update manually code as checkstyle is demanding, it will be changed back to what you have by latest Google formatter

**Comment by Aziz-755** [comment]
I am on it!

**Comment by romani** [comment]
@Aziz-755, please confirm that issue is valid, I still think it might be gray area of formatter and it might ignore both variants.

Issue is only valid if code is violated by checkstyle, user  change code as checkstyle demands, execution of formatter bring code back to original state.

**Comment by christophpre** [comment]
I just checked again. I was not using the default google formatter before, but the built-in Intellij formatter (default IDE code style). Sry for the confusion. Still, the issues persists. Should I create a separate ticket?

**Comment by larskristianhaga** [comment]
Having the same issue, worked fine on `10.23.1`, but breaks when upgrading to `10.24.0` even tho no other code was changed.

**Comment by romani** [comment]
Issue is valid, I took close look, chekcstyle demand switch to be moved more to left, that is nonsense.

**Comment by Aziz-755** [comment]
@romani Yes, I confirm it is a valid issue. I'll have a look at it. I suspect that it existed before, but since we were not producing violations for Lambdas, it was not visible

**Comment by romani** [comment]
Yes, most likely, we skipped raising violations on code we were not able to handle properly to avoid anger from users.
Now we become a bit more brave on such cases, and got immediate feedback/anger from users. We need to find way to fix it forward, we are ok to skip validating complicated cases.
False positives that we produce, blocks users to update checkstyle version.

**Comment by Aziz-755** [comment]
@christophpre I think the violations provided by checkstyle here [comment](https://github.com/checkstyle/checkstyle/issues/17120#issuecomment-2901664514) are valid. In your config.xml file this was the configuration for the indentation check : 
```
    <module name="Indentation">
      <property name="basicOffset" value="4"/>
      <property name="braceAdjustment" value="0"/>
      <property name="caseIndent" value="4"/>
      <property name="throwsIndent" value="8"/>
      <property name="lineWrappingIndentation" value="8"/>
      <property name="arrayInitIndent" value="4"/>
    </module>
``` 
the `lineWrappingIndentation` value is `8` and that's why checkstyle is complaining

**Comment by romani** [comment]
@christophpre , this conversation on non Google style config of Indentation, becomes confusing.
We will merge https://github.com/checkstyle/checkstyle/pull/17138
And please create issue by usage of snapshot version of checkstyle to show problem on your config.

**Comment by mohitsatr** [comment]
with config:
```
    <module name="Indentation">
      <property name="basicOffset" value="4"/>
      <property name="braceAdjustment" value="0"/>
      <property name="caseIndent" value="4"/>
      <property name="throwsIndent" value="8"/>
      <property name="lineWrappingIndentation" value="8"/>
      <property name="arrayInitIndent" value="4"/>
    </module>
```
```
$ cat OneClass.java
/** some data. */
public class OneClass {
    private static String switchIt(AnEnum value) {
        return switch (value) {
            case VALUE_1,
                    VALUE_2,
                    VALUE_3 -> "Test";
            default -> "Default";
        };
    }

    private enum AnEnum {
        VALUE_1,
        VALUE_2,
        VALUE_3
    }
}


$ java -jar testing/google-java-format-1.27.0-all-deps.jar OneClass.java
/** some data. */
public class OneClass {
  private static String switchIt(AnEnum value) {
    return switch (value) {
      case VALUE_1, VALUE_3, VALUE_4, VALUE_5, VALUE_6 -> "Test";
      default -> "Default";
    };
  }

  private enum AnEnum {
    VALUE_1,
    VALUE_2,
    VALUE_3
  }
}

$ java -jar testing/checkstyle-10.25.0-all.jar -c testing/google_checks.xml FormattedCode.java 
Starting audit...
Audit done.
```
Google-formatter just put all case options in one line.

**Comment by romani** [comment]
Please try same structure of code but with long names or a lot of  enums elements to not fit single line

**Comment by maistrovyi** [comment]
Issue still present

**Comment by larskristianhaga** [comment]
Same her, issue still presists

**Comment by mohitsatr** [comment]
```
$ cat OneClass.java

/** some data. */
public class OneClass {
    private static String switchIt(AnEnum value) {
        return switch (value) {
            case VALUE_MIXED_SYTLE_SWITCH_STATEMENT_1,
                    VALUE_MIXED_SYTLE_SWITCH_STATEMENT_2,
                    VALUE_MIXED_SYTLE_SWITCH_STATEMENT_3 -> "go to default statego to default statem";
            case VALUE_MIXED_SYTLE_SWITCH_STATEMENT_4,
                    VALUE_MIXED_SYTLE_SWITCH_STATEMENT_5,
                    VALUE_MIXED_SYTLE_SWITCH_STATEMENT_6 -> "go to default";
            default -> "Default";
        };
    }

    private enum AnEnum {
        VALUE_MIXED_SYTLE_SWITCH_STATEMENT_1,
        VALUE_MIXED_SYTLE_SWITCH_STATEMENT_2,
        VALUE_MIXED_SYTLE_SWITCH_STATEMENT_3,
        VALUE_MIXED_SYTLE_SWITCH_STATEMENT_4,
        VALUE_MIXED_SYTLE_SWITCH_STATEMENT_5,
        VALUE_MIXED_SYTLE_SWITCH_STATEMENT_6,
    }
}

$ java -jar testing/google-java-format-1.27.0-all-deps.jar OneClass.java 

/** some data. */
public class OneClass {
  private static String switchIt(AnEnum value) {
    return switch (value) {
      case VALUE_MIXED_SYTLE_SWITCH_STATEMENT_1,
          VALUE_MIXED_SYTLE_SWITCH_STATEMENT_2,
          VALUE_MIXED_SYTLE_SWITCH_STATEMENT_3 ->
          "go to default statego to default statem";
      case VALUE_MIXED_SYTLE_SWITCH_STATEMENT_4,
          VALUE_MIXED_SYTLE_SWITCH_STATEMENT_5,
          VALUE_MIXED_SYTLE_SWITCH_STATEMENT_6 ->
          "go to default";
      default -> "Default";
    };
  }

  private enum AnEnum {
    VALUE_MIXED_SYTLE_SWITCH_STATEMENT_1,
    VALUE_MIXED_SYTLE_SWITCH_STATEMENT_2,
    VALUE_MIXED_SYTLE_SWITCH_STATEMENT_3,
    VALUE_MIXED_SYTLE_SWITCH_STATEMENT_4,
    VALUE_MIXED_SYTLE_SWITCH_STATEMENT_5,
    VALUE_MIXED_SYTLE_SWITCH_STATEMENT_6,
  }
}
$ java -jar testing/google-java-format-1.27.0-all-deps.jar OneClass.java > FormattedCode.java
$ java -jar testing/checkstyle-10.25.0-all.jar -c testing/google_checks.xml FormattedCode.java 
Starting audit...
Audit done.
```
format is consistent with Google-formatter. so issue is not related to google-style

**Comment by romani** [comment]
@maistrovyi , @larskristianhaga , please create new issue and make sure you use latest released version
